src/test/groovy/pl/allegro/tech/build/axion/release/infrastructure/git/GitRepositoryTest.groovy

```diff
@@ -146,6 +148,31 @@ class GitRepositoryTest extends Specification {
         then:
         tags.tags == ['release-1']
     }
+    
+    def "should return all tagged commits matching the pattern provided"() {
+        given:
+        repository.tag('release-1')
+        repository.commit(['*'], "commit after release-1")
+        repository.tag('release-2')
+        repository.commit(['*'], "commit after release-2")
+        repository.tag('another-tag-1')
+        repository.commit(['*'], "commit after another-tag-1")
+        repository.commit(['*'], "commit after another-tag-1-2")
+        repository.tag('release-4')
+        repository.commit(['*'], "commit after release-4")
+        repository.tag('release-3')
+        repository.commit(['*'], "commit after release-3")
+    
+        when:
+        List<TagsOnCommit> allTaggedCommits  = repository.allTaggedCommits(~/^release.*/, null, true, false)
+    
+        then:
+        allTaggedCommits.size() == 4
+        allTaggedCommits.get(0).tags.get(0).equals('release-3')
```

Comment:
It didn't work.

    allTaggedCommits == ['release-3', 'release-4']
    |                |
    |                false
    [pl.allegro.tech.build.axion.release.domain.scm.TagsOnCommit@28daa04, pl.allegro.tech.build.axion.release.domain.scm.TagsOnCommit@5caeb946, pl.allegro.tech.
build.axion.release.domain.scm.TagsOnCommit@1d6e6513, pl.allegro.tech.build.axion.release.domain.scm.TagsOnCommit@1b04c147]
        at pl.allegro.tech.build.axion.release.infrastructure.git.GitRepositoryTest.should return all tagged commits matching the pattern provided(GitRepository
Test.groovy:175)